Tests/SourceGenerators/SourceGeneratorAssemblyToProcess/Generated/Realm.SourceGenerator/Realms.SourceGenerator.RealmGenerator/AllTypesClass_generated.cs

```diff
@@ -0,0 +1,259 @@
+                {
+                    UnsubscribeFromNotifications();
+                }
+            }
+        }
+
+        partial void OnPropertyChanged(string propertyName);
+
+        private void RaisePropertyChanged([CallerMemberName] string propertyName = null)
+        {
+            _propertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
+            OnPropertyChanged(propertyName);
+        }
+
+        private void SubscribeForNotifications()
+        {
+            Accessor.SubscribeForNotifications(RaisePropertyChanged);
+        }
+
+        private void UnsubscribeFromNotifications()
+        {
+            Accessor.UnsubscribeFromNotifications();
+        }
+
+        public static explicit operator AllTypesClass(RealmValue val) => val.AsRealmObject<AllTypesClass>();
+
+        public static implicit operator RealmValue(AllTypesClass val) => RealmValue.Object(val);
+
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public TypeInfo GetTypeInfo() => Accessor.GetTypeInfo(this);
+
+        public override bool Equals(object obj)
+        {
+            if (obj is null)
+            {
+                return false;
+            }
+
+            if (ReferenceEquals(this, obj))
+            {
+                return true;
+            }
+
+            if (obj is InvalidObject)
+            {
+                return !IsValid;
+            }
+
+            if (obj is not IRealmObjectBase iro)
+            {
+                return false;
+            }
+
+            return Accessor.Equals(iro.Accessor);
+        }
+
+        public override int GetHashCode() => IsManaged ? Accessor.GetHashCode() : base.GetHashCode();
+
+        public override string ToString() => Accessor.ToString();
+
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        private class AllTypesClassObjectHelper : IRealmObjectHelper
+        {
+            public void CopyToRealm(IRealmObjectBase instance, bool update, bool skipDefaults)
+            {
+                throw new InvalidOperationException("This method should not be called for source generated classes.");
+            }
+
+            public ManagedAccessor CreateAccessor() => new AllTypesClassManagedAccessor();
+
+            public IRealmObjectBase CreateInstance() => new AllTypesClass();
+
+            public bool TryGetPrimaryKeyValue(IRealmObjectBase instance, out object value)
+            {
+                value = null;
+                return false;
+            }
+        }
+    }
+}
+
+namespace SourceGeneratorPlayground.Generated
+{
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    internal interface IAllTypesClassAccessor : IRealmAccessor
+    {
+        char CharProperty { get; set; }
+    }
+
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    internal class AllTypesClassManagedAccessor : ManagedAccessor, IAllTypesClassAccessor
+    {
+        public char CharProperty
+        {
+            get => (char)GetValue("CharProperty");
```

Comment:
Should we use `nameof` here? I know we're generating it and is probably irrelevant, but might look nicer?

Reply:
Good point.